apis/kueue/v1beta1/provisioningrequestconfig_types.go

```diff
@@ -57,6 +58,58 @@ type ProvisioningRequestConfigSpec struct {
 	// +listType=set
 	// +kubebuilder:validation:MaxItems=100
 	ManagedResources []corev1.ResourceName `json:"managedResources,omitempty"`
+
+	// retryStrategy defines strategy for retrying ProvisioningRequest.
+	// If null, then the default configuration is applied with the following parameter values:
+	// BackoffLimitCount:  3
+	// BackoffBaseSeconds: 60 - 1 min
+	// BackoffMaxSeconds:  1800 - 30 mins
+	//
+	// To switch off retry mechanism
+	// set retryStrategy.backoffLimitCount to 0.
+	//
+	// +optional
+	RetryStrategy *ProvisioningRequestRetryStrategy `json:"retryStrategy,omitempty"`
+}
+
+var (
+	DefaultRetryStrategy = ProvisioningRequestRetryStrategy{
+		BackoffLimitCount:  ptr.To(int32(3)),
```

Comment:
Both are correct, the `DefaultRetryStrategy` is applied when a user doesn't specify `retryStrategy` field in ProvisioningRequestConfig. However, if they specify the `retryStrategy` field, but not `backoffLimitCount`, then it's null and workloads will endlessly requeue

Reply:
I see.
In that case, I think that we should handle the bahavior in the defaulter webhook.
Because the backoffLimitCount default value is dynamic. Such dynamic defaulting should be handled in the defaulter webhook instead of static (marker) and controller defaulting.

But, for other fields of the retryStrategy like backoffBaseSeconds, we can still handle as a static defaulting (marker).

Reply:
Discussed offline, current behavior is to always set default to 3 so it's not dynamic anymore